### PR TITLE
Add support for more recent JDKs

### PR DESCRIPTION
**Pull request by pierre**
Note that in order to use a more recent JVM, plugins must be re-compiled to relax the `(&(osgi.ee=JavaSE)(version=1.8))` requirement.

/cc @andrenpaes

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      uses: actions/setup-java@v2
+      with:
+        distribution: ${{ matrix.java-distribution }}
+        java-version: ${{ matrix.java-version }}
+    - name: Configure Sonatype mirror
+      uses: s4u/maven-settings-action@v2.3.0
+      # Go to Sonatype directly to avoid delay syncs (could get rid of this if actions/setup-java were to support mirrors).
+      with:
+        mirrors: '[{"id": "oss-releases", "name": "Sonatype releases", "mirrorOf": "*", "url": "https://oss.sonatype.org/content/repositories/releases/"}]'
+    - name: Check if killbill-oss-parent SNAPSHOT must be fetched
```
Comment (sbrossie) [inline .github/workflows/codeql-analysis.yml:43]: I guess I am confused on all the steps necessary for `SNAPSHOT` - do we need this ?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        repository: killbill/killbill-client-java
+        ref: ${{ github.ref }}
+        path: killbill-client-java
+    - name: Build killbill-client-java
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      run: |
+        cd $GITHUB_WORKSPACE/killbill-client-java
+        mvn ${MAVEN_FLAGS} clean install -DskipTests=true
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
```
Comment (sbrossie) [inline .github/workflows/codeql-analysis.yml:133]: Is this step where we add the code into the `codeQL` database -- or does this happen automatically when we run the build?

**Review by sbrossie — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,144 @@
+name: "CodeQL"
+
+on:
+  - push
```
[the remark this answers is not in the document]
Comment (pierre) [reply .github/workflows/codeql-analysis.yml:4]: I don't see a downside of running it on each push? This simply updates the (private) security tab.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (pierre) [reply .github/workflows/codeql-analysis.yml:43]: Yes :-) I can fill you in IRL.

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (pierre) [reply .github/workflows/codeql-analysis.yml:133]: The checkout happens here: https://github.com/killbill/killbill/pull/1462/files/1997e0a7c0f3ec497c1f15e4e023fd0eb0774556#diff-63bd641104d10e25f141d518a16b22a151d125e12701df2f9e79734b23b90188R20

**Pull request merged**